### PR TITLE
Add Ritual bag

### PR DESCRIPTION
**Pull request by raphaelzoellner**
ritual_bag_throw

### CODE REVIEW

File: src/main/kotlin/com/reicheltp/celtic_rituals/init/ModBlocks.kt [hunk 1]
```diff
@@ -10,7 +9,4 @@ object ModBlocks {
 
     @ObjectHolder("celtic_rituals:ritual_bowl")
     var RITUAL_BOWL: Block? = null
-
-    @ObjectHolder("celtic_rituals:ritual_bowl")
-    var RITUAL_BOWL_TILE: TileEntityType<*>? = null
```
Comment (reicheltp) [inline src/main/kotlin/com/reicheltp/celtic_rituals/init/ModBlocks.kt]: Why is this moved to another file? This belongs to the block, and TileEntity is not an Entity

File: src/main/kotlin/com/reicheltp/celtic_rituals/init/ModEntities.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.reicheltp.celtic_rituals.rituals.bag.RitualBagEntity
+import net.minecraft.entity.EntityClassification
+import net.minecraft.entity.EntityType
+import net.minecraft.tileentity.TileEntityType
+import net.minecraftforge.registries.ObjectHolder
+
+object ModEntities {
+    @ObjectHolder("celtic_rituals:ritual_bag_entity")
+    var RITUAL_BAG_ENTITY: EntityType<RitualBagEntity> =
```
Comment (reicheltp) [inline src/main/kotlin/com/reicheltp/celtic_rituals/init/ModEntities.kt]: Use ObjectHolder instead of creating the entity. This allows other to override this.

File: src/main/kotlin/com/reicheltp/celtic_rituals/rituals/bag/RitualBagEntity.kt [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            return ItemStack(ModItems.RITUAL_BAG)
+        } else {
+            return itemstack
+        }
+    }
+
+    // this needs to be overriden for custom non living entities to render also see:
+    // https://www.minecraftforge.net/forum/topic/72657-solved-1143-entity-renderer-not-rendering/
+    // https://www.minecraftforge.net/forum/topic/75827-solved1144-problem-with-custom-entity-rendering-i-guess/
+    // https://www.minecraftforge.net/forum/topic/71717-solved-1142-register-entities-and-rendering/
```
Comment (reicheltp) [inline src/main/kotlin/com/reicheltp/celtic_rituals/rituals/bag/RitualBagEntity.kt]: Please use kdoc format when comment functions. `/** .. */

File: src/main/kotlin/com/reicheltp/celtic_rituals/rituals/bag/RitualBagItem.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        private val LOGGER = LogManager.getLogger()
+    }
+
+    init {
+        registryName = ResourceLocation(MOD_ID, "ritual_bag")
+    }
+
+    override fun onItemRightClick(worldIn: World, playerIn: PlayerEntity, handIn: Hand): ActionResult<ItemStack> {
+        val itemstack = playerIn.getHeldItem(handIn)
+        val itemstack1 = if (playerIn.abilities.isCreativeMode) itemstack.copy() else itemstack.split(1)
```
Comment (reicheltp) [inline src/main/kotlin/com/reicheltp/celtic_rituals/rituals/bag/RitualBagItem.kt]: We should not have custom creative code as this can lead to possible bugs.

**Comment by reicheltp** [comment]
Alternative Bag design
image

And colored (color is picked corresponding to bound ritual)
image

**Comment by Sebkasanzew** [comment]
> Alternative Bag design
> image
> 
> And colored (color is picked corresponding to bound ritual)
> image

This doesn't look like vomit at least 😄
